### DISCUSSION TITLE
How to add custom attribute to Identity Provider?

### DISCUSSION THREAD

**Discussion by alirf81** [keycloak/keycloak#14191] [Q&A] [answered]
Good day!
I need to add custom attribute to the Identity Provider model and update Keycloak theme to make it available to edit the attribute in admin panel.
I would appreciate for any ideas of how to implement this.

**Comment by lexcao** [comment, marked as answer]
For a custom attribute to the Identity provider,
You could extends `OIDCIdentityProviderConfig` or `OAuth2IdentityProviderConfig` to create your own config.

For the theme on the legacy admin console.
You could refer to this [doc](https://www.keycloak.org/docs/latest/server_development/index.html#_themes)

For the new admin console,
It is just finished development, but not released,  please refer to [this](https://github.com/keycloak/keycloak-ui/pull/3167)

**Reply by alirf81** [reply]
Thank you for your comment!
